plugins/package-managers/spdx/src/main/kotlin/SpdxDocumentFile.kt

```diff
@@ -336,26 +336,36 @@ class SpdxDocumentFile(
     /**
      * Return the dependencies of the package with the given [pkgId] defined in [doc] of the
      * [SpdxRelationship.Type.DEPENDENCY_OF] type. Identified dependencies are mapped to ORT [Package]s and then
-     * added to [packages].
+     * added to [packages]. The [parents] set contains the IDs of the package that have already been encountered; it
+     * is used to detect circular dependencies.
      */
     private fun getDependencies(
         pkgId: String,
         doc: SpdxResolvedDocument,
-        packages: MutableSet<Package>
-    ): Set<PackageReference> =
-        getDependencies(pkgId, doc, packages, SpdxRelationship.Type.DEPENDENCY_OF) { target ->
+        packages: MutableSet<Package>,
+        parents: MutableSet<String>
+    ): Set<PackageReference> {
+        logger().debug { "Retrieving dependencies for package '$pkgId'." }
+
+        if (!parents.add(pkgId)) {
+            logger().warn { "A cycle was detected in the dependencies for packages $parents." }
```

Comment:
Should this additional say: "Not adding $pkgId." ?

Reply:
If `add` returns *false*, this means that this ID is already contained in the set, so it is listed in the output.